students/YingGuo/lession07/html_render.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element():
+    tag_name = "html"
+    indent = "    "
+
+    def __init__(self, content=None, **kwargs):
+        if content is None:
+            self.content = []
+        if content is not None:
+            self.content = [content]
+        self.kwargs = kwargs 
+
+
+    def append(self, new_content):
+        self.content.append(new_content)
+
+    def write_html(self):
+        with open("output_html","w") as f:
+             f.write(self.content)
+             
+    #what does ind mean in the test_html? why it is needed?
+    def render(self, out_file, cur_ind=""):
+        if cur_ind:
+            out_file.write(cur_ind)
+        else:
+            pass
```

Comment:
Minor: No need to put `else: pass`. It's of no use, and that's exactly what happens when you don't enter that code.